301d58/code-challenges/regex/challenges-04.test.js

```diff
@@ -0,0 +1,236 @@
+CHALLENGE 2
+
+Write a function named isCapitalized that takes in a string. This function should use a regular expression pattern to match all words that begin with a capital letter. It should only match words, not punctuation.
+
+Return an array containing all the matches.
+------------------------------------------------------------------------------------------------ */
+
+const isCapitalized = ( str ) => {
+  let temp;
+  let capWords = [];
+  let arr = str.split( " " );
+  arr.forEach(element => {
+    if (( /^[A-Z]/ ).test( element )){
+      temp = element.replace(/[(,).]/g, '' );
+      capWords.push( temp );
+    }
+  });
+  return capWords;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+------------------------------------------------------------------------------------------------ */
+
+const citiesAtoJ = ( arr ) => {
+  let arrTemp = [];
+  arr.forEach(element => {
+    if ( ( /(^[A-J])/g ).test( element ) ){
+      arrTemp.push( element );
+    }
+  });
+  return arrTemp;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+You have created a game application and begin by asking users an easy question: In which month is Halloween?
+
+Write a function named matchMonth which uses a regular expression pattern to match any of these inputs: October, Oct, october, oct
+
+If the user enters any of these four inputs, return true. For any other input, return false.
+
+Do not use the vertical bar (pipe) in your pattern.
+------------------------------------------------------------------------------------------------ */
+
+const matchMonth = ( input ) => {
+  if ( ( /^[Oo]ct(ober)?$/g ).test( input ) ){
+    return true;
+  }
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named noPunctuation that contains a regular expression pattern to find all of the words that contain a space immediately at the end of the word. Return an array of all such words, still containing the space at the end.
+
+For example, if given the string "Hello, and have a wonderful day!", the word "Hello, " would not be returned because it is immediately followed by a comma. The word "day!" would not be returned because it is immediately followed by an exclamation point.
+
+The expected output of "Hello, and have a wonderful day!" is ["and ", "have ", "a ", "wonderful "].
+------------------------------------------------------------------------------------------------ */
+
+const noPunctuation = str => {
+  // Need to figure a way to hold the space in the array.
```

Comment:
Some of the methods discussed in class with return the part of the string that the Regex finds